packages/origin-247-claim/src/repositories/MatchResult/MatchResultPostgres.repository.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository, InjectConnection } from '@nestjs/typeorm';
+import { Repository, Connection, Between, In, createQueryBuilder } from 'typeorm';
+
+import { MatchResultEntity, tableName } from './MatchResult.entity';
+import {
+    FindOptions,
+    MatchResultRepository,
+    NewMatchResult,
+    Entity,
+    MatchResultColumns,
+    GroupedMatchResults
+} from './MatchResult.repository';
+
+@Injectable()
+export class MatchResultPostgresRepository implements MatchResultRepository {
+    constructor(
+        @InjectRepository(MatchResultEntity)
+        private repository: Repository<MatchResultEntity>,
+        @InjectConnection()
+        private connection: Connection
+    ) {}
+
+    public async create(matchResult: NewMatchResult): Promise<MatchResultEntity> {
+        return await this.repository.save(matchResult);
+    }
+
+    public async getAll(): Promise<MatchResultEntity[]> {
+        return await this.repository.find();
+    }
+
+    public async find(findOptions: FindOptions): Promise<MatchResultEntity[]> {
+        return await this.repository.find({
+            where: {
+                consumerId: In(findOptions.consumerIds),
+                generatorId: In(findOptions.generatorIds),
+                from: Between(findOptions.from, findOptions.to),
+                to: Between(findOptions.from, findOptions.to)
+            }
+        });
+    }
+
+    public async findGrouped(
+        findOptions: FindOptions,
+        groupOptions: Entity[]
```

Comment:
TBH I don't understand how this API works here.

Reply:
Maybe changing the names to filterBy and groupBy would make more sense?
findOptions is the data for **WHERE**
` 
example:
    {
        consumerIds: ['consumerOne', 'consumerTwo'],
        generatorIds: ['generatorOne', 'generatorTwo', 'generatorThree'],
        from: new Date('2021-08-01T10:00:00.000Z'),
        to: new Date('2021-08-01T12:00:00.000Z')
    }
`
groupOptions is the data for **GROUP BY**
`
examples:
groupOptions = [Entity.Generator];
groupOptions = [[Entity.Generator, Entity.Consumer];
groupOptions = [[Entity.Consumer, Entity.Generator];
`

Reply:
Oh, so it's wrong data typa I think. It should be `[Entity] | [Entity, Entity]` ?